src/NNlib.jl

```diff
@@ -13,11 +13,27 @@ using LinearAlgebra
 using LinearAlgebra.BLAS: @blasfunc, BlasInt
 using LinearAlgebra: AdjOrTransAbsMat, Adjoint, BlasFloat, Transpose
 using Random
+using ScopedValues
 using Statistics
 using Statistics: mean
 
 const Numeric = Union{AbstractArray{<:T}, T} where {T<:Number}
 
+const ALLOW_SPAWNS = ScopedValue(true)
```

Comment:
Another thought: would it make sense to have this be an integer param that controls the max degree of spawn parallelism NNlib can use? Setting that param <= 1 (or to `false`) would disable spawning as before.

Reply:
That could make sense. I'll comment that this is internal, so it can be changed when that design is figured out.

Reply:
Yes, I should've elaborated on this thought. The idea would be to have a way to control the degree of parallelism via a context manager/macro like `disallow_spawns`, but using a numeric param instead of an on/off flag.

But this design can be fleshed out later. Then there are more QoL things (not for this PR) we can consider around threading that would complement it. For example, deciding the number of tasks to spawn based on # of threads / # of BLAS threads.